@tombentley The change looks good to me but there's a checkstyle issue:
```
[ant:checkstyle] [ERROR] /Users/mickael/github-ws/kafka/clients/src/test/java/org/apache/kafka/common/config/provider/DirectoryConfigProviderTest.java:103:85: ',' is not followed by whitespace. [WhitespaceAfter]
```

--- a/clients/src/test/java/org/apache/kafka/common/config/provider/DirectoryConfigProviderTest.java
+++ b/clients/src/test/java/org/apache/kafka/common/config/provider/DirectoryConfigProviderTest.java
@@ -100,7 +100,7 @@ public class DirectoryConfigProviderTest {
     @Test
     public void testNoSubdirs() {
         // Only regular files directly in the path directory are allowed, not in subdirs
-        Set<String> keys = toSet(asList(subdir.getName(), String.join(File.separator,subdir.getName(), subdirFile.getName())));
+        Set<String> keys = toSet(asList(subdir.getName(), String.join(File.separator, subdir.getName(), subdirFile.getName())));
         ConfigData configData = provider.get(dir.getAbsolutePath(), keys);
         assertTrue(configData.data().isEmpty());
         assertNull(configData.ttl());

